include/z64cutscene.h

```diff
@@ -3,286 +3,501 @@
+    /* 0x4F */ CS_DEST_LON_LON_RANCH_1, // unused
+    /* 0x50 */ CS_DEST_LON_LON_RANCH_2, // unused
+    /* 0x51 */ CS_DEST_LON_LON_RANCH_3, // unused
+    /* 0x52 */ CS_DEST_LON_LON_RANCH_4, // unused
+    /* 0x53 */ CS_DEST_LON_LON_RANCH_5, // unused
+    /* 0x54 */ CS_DEST_LON_LON_RANCH_6, // unused
+    /* 0x55 */ CS_DEST_LON_LON_RANCH_7, // unused
+    /* 0x56 */ CS_DEST_LON_LON_RANCH_8, // unused
+    /* 0x57 */ CS_DEST_LON_LON_RANCH_9, // unused
+    /* 0x58 */ CS_DEST_LON_LON_RANCH_10, // unused
+    /* 0x59 */ CS_DEST_LON_LON_RANCH_11, // unused
+    /* 0x5A */ CS_DEST_LON_LON_RANCH_12, // unused
+    /* 0x5B */ CS_DEST_LON_LON_RANCH_13, // unused
+    /* 0x5C */ CS_DEST_LON_LON_RANCH_14, // unused
+    /* 0x5D */ CS_DEST_LON_LON_RANCH_15, // unused
+    /* 0x5E */ CS_DEST_LON_LON_RANCH_FROM_EPONAS_SONG,
+    /* 0x5F */ CS_DEST_STONES_RESTORED_CONDITIONAL, // unused
+    /* 0x60 */ CS_DEST_DESERT_COLOSSUS_FROM_BLUE_WARP,
+    /* 0x61 */ CS_DEST_GRAVEYARD_FROM_BLUE_WARP,
+    /* 0x62 */ CS_DEST_DEATH_MOUNTAIN_CRATER_FROM_BLUE_WARP,
+    /* 0x63 */ CS_DEST_SACRED_FOREST_MEADOW_FROM_BLUE_WARP,
+    /* 0x64 */ CS_DEST_KOKIRI_FOREST_FROM_BLUE_WARP,
+    /* 0x65 */ CS_DEST_DESERT_COLOSSUS_FROM_NABOORU_CAPTURE,
+    /* 0x66 */ CS_DEST_TEMPLE_OF_TIME_FRONT_OF_PEDESTAL, // unused
+    /* 0x67 */ CS_DEST_HYRULE_FIELD_TITLE_SCREEN,
+    /* 0x68 */ CS_DEST_TITLE_SCREEN_DEMO,
+    /* 0x69 */ CS_DEST_GRAVEYARD_SUNS_SONG_PART_2,
+    /* 0x6A */ CS_DEST_ROYAL_FAMILYS_TOMB_SUNS_SONG_PART_3,
+    /* 0x6B */ CS_DEST_GANONS_CASTLE_DISPEL_FOREST_BEAM,
+    /* 0x6C */ CS_DEST_GANONS_CASTLE_DISPEL_WATER_BEAM,
+    /* 0x6D */ CS_DEST_GANONS_CASTLE_DISPEL_SHADOW_BEAM,
+    /* 0x6E */ CS_DEST_GANONS_CASTLE_DISPEL_FIRE_BEAM,
+    /* 0x6F */ CS_DEST_GANONS_CASTLE_DISPEL_LIGHT_BEAM,
+    /* 0x70 */ CS_DEST_GANONS_CASTLE_DISPEL_SPIRIT_BEAM,
+    /* 0x71 */ CS_DEST_GANONS_CASTLE_DISPEL_BARRIER_CONDITONAL,
+    /* 0x72 */ CS_DEST_HYRULE_FIELD,
+    /* 0x73 */ CS_DEST_HYRULE_FIELD_FROM_IMPA_ESCORT,
+    /* 0x74 */ CS_DEST_FROM_RARU_CONDITIONAL,
+    /* 0x75 */ CS_DEST_HYRULE_FIELD_CREDITS_SKY,
+    /* 0x76 */ CS_DEST_GANON_BATTLE_TOWER_COLLAPSE,
+    /* 0x77 */ CS_DEST_ZELDAS_COURTYARD_RECEIVE_LETTER
+} CutsceneDestination;
 
 typedef struct {
     /* 0x00 */ u16 base;
     /* 0x02 */ u16 startFrame;
     /* 0x04 */ u16 endFrame;
-} CsCmdBase; // size = 0x6
+    /* 0x06 */ u8 unused_06[0x2];
+} CsCmdGeneric; // size = 0x8
 
 typedef struct {
-    /* 0x00 */ u8  unk_00;
-    /* 0x01 */ u8  setting;
+    /* 0x00 */ u16 type;
     /* 0x02 */ u16 startFrame;
     /* 0x04 */ u16 endFrame;
-} CsCmdEnvLighting; // size = 0x6
+    /* 0x06 */ u8 unused_06[0x2A];
+} CsCmdMisc; // size = 0x30
 
 typedef struct {
-    /* 0x00 */ u8  unk_00;
-    /* 0x01 */ u8  sequence;
-    /* 0x02 */ u16 startFrame;
-    /* 0x04 */ u16 endFrame;
-} CsCmdMusicChange; // size = 0x6
+    /* 0x00 */ u8 unused_00[0x1];
+    /* 0x01 */ u8 settingPlusOne;
```

Comment:
seems that these should be u16 like the others

Reply:
it cannot due to lb/lh. This is the main issue with going with the "common struct" route. I believe a cast on the access would work, but its imo worse than just having a struct for each where we can name the fields however we want without them being generic

Reply:
It's more a question of whether `unused_00` is actually real; z_demo.c OKs fine for me when `settingPlusOne` is typed as a u16 and then cast to a (u8) when read

Reply:
that is what i said above yes-- I said a cast should work but its not really an improvement imo.
Going the common struct route means doing these casts, which makes things harder to follow and easier to mess up. 

One could see u16 in the struct and think their upper limit for this value is short max, when its really a byte. Having a struct for each command allows the variables to be typed precisely, and lets us give them descriptive names

Reply:
to illustrate my point:
what i have currently:
```c
typedef struct {
    /* 0x00 */ u8 unused_00[0x1];
    /* 0x01 */ u8 settingPlusOne;
    /* 0x02 */ u16 frame;
    /* 0x04 */ u8 unused_04[0x2C];
} CsCmdLightSetting; // size = 0x30
```
and if you wanted to use a common struct, like they probably originally did:
```
typedef struct {
    /* 0x00 */ u16 val;
    /* 0x02 */ u16 startFrame;
    /* 0x04 */ u16 endFrame;
    /* 0x06 */ Vec3us vec1;
    /* 0x0C */ Vec3i vec2;
    /* 0x18 */ Vec3i vec3;
    /* 0x24 */ Vec3i vec4;
} CsCmdLarge; // size = 0x30
```
with a common struct you cant name any of the variables in a meaningful way, because this struct has to apply to 6 different commands. even though in practice, only actor cues use any of the vectors. but now you miss out on the ability to type and name things as appropriate to the context of any given command.
At least with a struct for each command, you get meaningful names and types that properly reflect their usage, even if there is more padding in between.